spec/integration/puppet_x/puppetlabs/powershell_manager_spec.rb

```diff
@@ -165,6 +169,24 @@ class PowerShellManager; end
       expect(first_pid).to eq(second_pid)
     end
 
+    it "should create a new PowerShell manager host if user code exits the first process" do
+      first_pid = manager.execute('[Diagnostics.Process]::GetCurrentProcess().Id')[:stdout]
+      exitcode = manager.execute('[Diagnostics.Process]::GetCurrentProcess().Kill()')[:exitcode]
+
+      # when a process gets torn down out from under manager, there is no response
+      # a nil exitcode is a reasonable signal that the PS process was torn out from under it
+      expect(exitcode).to eq(nil)
```

Comment:
Decided that we want the PowerShellManager to deal with unexpected pipe breakages and to return a `-1` error code with some `stderr` indicating bad things happened.

That allows us to proceed on subsequent resource applications.